vlm/src/main/scala/geotrellis/contrib/vlm/avro/GeoTrellisDataPath.scala

```diff
@@ -0,0 +1,71 @@
+package geotrellis.contrib.vlm.avro
+
+import geotrellis.contrib.vlm.DataPath
+
+
+/** Represents a path that points to a GeoTrellis layer saved in a catalog.
+ *
+ *  @param path Path to the layer. This can be either an Avro or COG layer.
+ *    The given path needs to be in a `URI` format that include the following query
+ *    parameters:
+ *      - '''layer''': The name of the layer.
+ *      - '''zoom''': The zoom level to be read.
+ *      - '''band_count''': The number of bands of each Tile in the layer.
+ *    Of the above three parameters, `layer` and `zoom` are required. In addition,
+ *    this path can be prefixed with, '''gt+''' to signify that the target path
+ *    is to be read in only by [[GeotrellisRasterSource]].
+ *  @example "s3://bucket/catalog?layer=layer_name&zoom=10"
+ *  @example "hdfs://data-folder/catalog?layer=name&zoom-12&band_count=5"
+ *  @example "gt+file:///tmp/catalog?layer=name&zoom=5"
+ *  @note The order of the query parameters does not matter.
+ */
+case class GeoTrellisDataPath(val path: String) extends DataPath {
+  private val servicePrefix: String = "gt+"
+
+  private val layerNameParam: String = "layer"
+  private val zoomLevelParam: String = "zoom"
+  private val bandCountParam: String = "band_count"
+
+  require(path.contains(s"$layerNameParam="), s"The layer query parameter must be in the given path: $path")
+
+  // TODO: Support having the zoom parameter be optional
+  require(path.contains(s"$zoomLevelParam="), s"The zoom query parameter must be in the given path: $path")
```

Comment:
I wasn't sure if finding an appropriate zoom level if none was given fit in this PR or not, so I made it a requirement for right now.

Reply:
I'm worrying a lot about such an enormous amount of `require` calls everywhere, we have the same issue in GeoTrellis. I would like not at least to increase amount of these `require` calls.